py/desispec/workflow/desi_proc_funcs.py

```diff
@@ -715,6 +736,119 @@ def create_desi_proc_batch_script(night, exp, cameras, jobdesc, queue, runtime=N
 
     return scriptfile
 
+def create_desi_proc_tilenight_batch_script(night, tileid, queue, runtime=None, batch_opts=None,
+                                  system_name=None, mpistdstars=None, gpuspecter=None,
+                                  gpuextract=None,
+                                  ):
+    """
+    Generate a SLURM batch script to be submitted to the slurm scheduler to run desi_proc.
+
+    Args:
+        night: str or int. The night the data was acquired.
+        tileid: str or int. The tile id for the data.
+        queue: str. Queue to be used.
+
+    Options:
+        runtime: str. Timeout wall clock time.
+        batch_opts: str. Other options to give to the slurm batch scheduler (written into the script).
+        system_name: name of batch system, e.g. cori-haswell, cori-knl.
+        mpistdstars: bool. Whether to use MPI for stdstar fitting.
+        gpuspecter: bool. Whether to use gpu_specter.
+        gpuextract: bool. Whether to perform gpu extraction with gpu_specter.
+
+    Returns:
+        scriptfile: the full path name for the script written.
+
+    """
+
+    batchdir = get_desi_proc_batch_file_path(night)
+
+    os.makedirs(batchdir, exist_ok=True)
+
+    jobname = get_desi_proc_tilenight_batch_file_name(night, tileid)
+
+    timingfile = f'{jobname}-timing-$SLURM_JOBID.json'
+    timingfile = os.path.join(batchdir, timingfile)
+
+    scriptfile = os.path.join(batchdir, jobname + '.slurm')
+
+    batch_config = batch.get_config(system_name)
+    threads_per_core = batch_config['threads_per_core']
+    gpus_per_node = batch_config['gpus_per_node']
+
+    nodes = 1
+    if system_name == 'perlmutter-gpu':
+        ncores = 64
+    elif system_name == 'perlmutter-cpu':
+        ncores = 128
+    else:
+        raise RuntimeError('tilenight batch jobs only supported on Perlmutter')
+
+    if runtime is None:
+        runtime = 30
```

Comment:
Consider expanding `desispec.workflow.desi_proc_funcs.determine_resources` to support jobdesc=tilenight as well.  That is where other steps have their logic for determining what runtime to request as a function of number of exposures, system-name, etc.  e.g. runtime should scale with number of exposures, and the normal runtime vs. nexp is somewhat different for perlmutter-cpu vs. permutter-gpu (and the ratio will likely improve in the direction of perlmutter-gpu in the future).

Reply:
done.